src/modules/extrakeys/main_impl.h

```diff
@@ -155,6 +155,25 @@ int secp256k1_xonly_pubkey_tweak_add_check(const secp256k1_context* ctx, const u
             && secp256k1_fe_is_odd(&pk.y) == tweaked_pk_parity;
 }
 
+typedef struct {
+    const secp256k1_context *ctx;
+} secp256k1_xonly_sort_cmp_data;
+
+static int secp256k1_xonly_sort_cmp(const void* pk1, const void* pk2, void *cmp_data_tmp) {
+    secp256k1_xonly_sort_cmp_data* cmp_data = cmp_data_tmp;
+    return secp256k1_xonly_pubkey_cmp(cmp_data->ctx, *(secp256k1_xonly_pubkey **)pk1, *(secp256k1_xonly_pubkey **)pk2);
```

Comment:
nit: I think it would be more readable to simply cast the pointer (but I guess the compiler does not really do an assignment anyway).

Since the context pointer is a pointer, we could also save the `secp256k1_xonly_sort_cmp_data` type entirely and simply pass the context pointer a third argument to `secp256k1_xonly_sort_cmp`. Not sure if this is better, but it's less boilerplate at least.

Reply:
> I think it would be more readable

Hm, I found this slightly less readable, but either way is fine for me.

> we could also save the secp256k1_xonly_sort_cmp_data type

The reason the context is wrapped is that the `cmp_data` argument expects a non-const pointer and this seems slightly cleaner despite the boilerplate than having to cast the context's constness away.

Reply:
Added a comment why the struct exists.

Reply:
Oh sure, I missed the constness. In that case, the current version is cleaner for sure.